lib/services/firewall.pm

```diff
@@ -30,13 +30,14 @@ sub install_service {
 }
 
 sub susefirewall2_to_firewalld {
-    my $timeout = 180;
+    my $timeout = 360;
     assert_script_run('susefirewall2-to-firewalld -c',                                     timeout => $timeout);
     assert_script_run('firewall-cmd --permanent --zone=external --add-service=vnc-server', timeout => $timeout);
     # On some platforms such as Aarch64, the 'firewalld restart'
     # can't finish in the default timeout.
     systemctl 'restart firewalld', timeout => $timeout;
     script_run('iptables -S', timeout => $timeout);
+    set_var('SUSEFIREWALL2_SERVICE_CHECK', 1);
```

Comment:
I took a look and there is already the method "firewall" within opensusebasetest which returns the string 'SuSEfirewall2' or 'firewalld' depending on settings: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/opensusebasetest.pm#L1156

It seems that method is setting the expected firewall to firewalld if an upgrade is detected based on the HDD image name trying to match for `(1[123]|42)[\.-]` which only matches for openSUSE base images. Maybe it can work to simply extend that expression to match SLE upgrades as well. If not then still this method should be used to compute the expected firewall. At the very least this variable here could be evaluated in there.

Reply:
yes, we've changed that function with a var for service_check, which we changed susefirewall to firewalld.
sub firewall {
    my $old_product_versions      = is_sle('<15') || is_leap('<15.0');
    my $upgrade_from_susefirewall = is_upgrade && get_var('HDD_1') =~ /\b(1[123]|42)[\.-]/;
    return (($old_product_versions || $upgrade_from_susefirewall) && !is_tumbleweed && !(check_var('SUSEFIREWALL2_SERVICE_CHECK', 1))) ? 'SuSEfirewall2' : 'firewalld';
}

Reply:
@okurz  oliver can you take a look again.